emulators/mame/Portfile

```
@@ -66,20 +176,328 @@ build.args-append   ARCHOPTS="${configure.cxxflags}" \
+    foreach target ${g_mame_build_targets_main_list} {
+        ui_msg "Building main target: ${target}"
+        mame_build_run \
+            ${worksrcpath} \
+            ${target} \
+            ${build.args}
+    }
+
+    unset target
+}
+
+destroot {
+    set mame_target_dir "${destroot}${prefix}/libexec/mame"
+    set mame_share_dir "${destroot}${prefix}/share/mame"
+
+    # Main
+    xinstall -m 755 -d ${mame_target_dir}
+    copy ${worksrcpath}/${executable} ${mame_target_dir}
+    copy ${worksrcpath}/artwork ${mame_target_dir}
+    copy ${worksrcpath}/bgfx ${mame_target_dir}
+    xinstall -m 755 -d ${mame_target_dir}/cfg
+    xinstall -m 755 -d ${mame_target_dir}/cheat
+    xinstall -m 755 -d ${mame_target_dir}/comments
+    xinstall -m 755 -d ${mame_target_dir}/crsshair
+    copy ${worksrcpath}/ctrlr ${mame_target_dir}
+    xinstall -m 755 -d ${mame_target_dir}/diff
+    xinstall -m 755 -d ${mame_target_dir}/fonts
+    copy ${worksrcpath}/hash ${mame_target_dir}
+    copy ${worksrcpath}/hlsl ${mame_target_dir}
+    copy ${worksrcpath}/ini ${mame_target_dir}
+    xinstall -m 755 -d ${mame_target_dir}/inp
+    copy ${worksrcpath}/keymaps ${mame_target_dir}
+    copy ${worksrcpath}/language ${mame_target_dir}
+    xinstall -m 755 -d ${mame_target_dir}/nvram
+    copy ${worksrcpath}/plugins ${mame_target_dir}
+    copy ${worksrcpath}/roms ${mame_target_dir}
+    copy ${worksrcpath}/samples ${mame_target_dir}
+    copy ${worksrcpath}/scripts ${mame_target_dir}
+    xinstall -m 755 -d ${mame_target_dir}/software
+    xinstall -m 755 -d ${mame_target_dir}/snap
+    xinstall -m 755 -d ${mame_target_dir}/sta
+
+    # Docs
+    xinstall -m 755 -d ${mame_share_dir}
+    copy ${worksrcpath}/docs/build/man ${mame_share_dir}
+
+    # Tools
+    if {[variant_isset tools]} {
+        xinstall -m 755 \
+            ${worksrcpath}/aueffectutil \
+            ${worksrcpath}/castool \
+            ${worksrcpath}/chdman \
+            ${worksrcpath}/floptool \
+            ${worksrcpath}/imgtool \
+            ${worksrcpath}/jedutil \
+            ${worksrcpath}/ldresample \
+            ${worksrcpath}/ldverify \
+            ${worksrcpath}/nltool \
+            ${worksrcpath}/nlwav \
+            ${worksrcpath}/pngcmp \
+            ${worksrcpath}/regrep \
+            ${worksrcpath}/romcmp \
+            ${worksrcpath}/split \
+            ${worksrcpath}/srcclean \
+            ${worksrcpath}/testkeys \
+            ${worksrcpath}/unidasm \
+            ${mame_target_dir}
+    }
+
+    unset mame_share_dir
+    unset mame_target_dir
+}
+
+post-destroot {
+    set mp_man_dest_dir "${destroot}${prefix}/share/man/man1"
+    set mame_target_dir "${destroot}${prefix}/libexec/mame"
+    set mame_share_dir "${destroot}${prefix}/share/mame"
+    set mame_launch_script "${mame_target_dir}/mame.sh"
+    set mame_launch_link "${destroot}${prefix}/bin/mame"
+
+    # Create launch wrapper script
+    xinstall -m 755 "${distpath}/mame-script-template-launcher.sh" \
```

Destrooting the port failed:
```
Error: Failed to destroot mame: xinstall: Cannot stat: /opt/local/var/macports/distfiles/mame/mame-script-template-launcher.sh, No such file or directory
```
The file is not downloaded by MacPorts to distpath; you're providing it in filespath:
```suggestion
    xinstall ${filespath}/mame-script-template-launcher.sh \
```
(0755 is the default mode.)